client/core/core.go

```diff
@@ -2932,6 +2934,61 @@ func (c *Core) tipChange(assetID uint32, nodeErr error) {
 	c.updateBalances(counts)
 }
 
+func (c *Core) PromptShutdown() bool {
+	c.connMtx.Lock()
+	defer c.connMtx.Unlock()
+	ok := true
+	for _, dc := range c.conns {
+		if dc.hasActiveOrders() {
+			ok = false
+			break
+		}
+	}
+	if !ok {
+		fmt.Print("You have active orders. Shutting down now may result in failed swaps and account penalization. " +
+			"Do you want to quit anyway? ('y' to quit, enter to abort):")
+		scanner := bufio.NewScanner(os.Stdin)
+		scan := make(chan bool)
+		go func() {
+			scan <- scanner.Scan()
+		}()
+		select {
+		case <-c.ctx.Done():
+			return ok
+		case <-scan:
+		}
+
+		promptResp := scanner.Text()
+		err := scanner.Err()
+		if err != nil {
+			fmt.Printf("input failed: %v", err)
+			return ok
+		}
+
+		switch promptResp {
```

Comment:
See previous comment https://github.com/decred/dcrdex/pull/527#discussion_r450893004. Can drop `prompResp` variable and just do 

`switch strings.ToLower(scanner.Text()) {`

Reply:
Good point.  Only `scanner.Scan` can generate an error that `scanner.Err` would report, not `scanner.Text`, so you might as well put it in the switch.